scripts/zones/Arrapago_Remnants/mobs/Qiqirn_Astrologer.lua

```diff
@@ -0,0 +1,86 @@
+-----------------------------------
+-- MOB: Qiqirn Astrologer
+-- Area: Arrapago Remnants
+-----------------------------------
+require("scripts/globals/monstertpmoves")
+require("scripts/globals/status")
+require("scripts/globals/msg")
+require("scripts/globals/teleports")
+require("scripts/globals/pathfind")
+local ID = require("scripts/zones/Arrapago_Remnants/IDs")
+-----------------------------------
+
+function onMobSpawn(mob)
+   mob:setMobMod(dsp.mobMod.HP_STANDBACK, -1) 
+end
+
+
+
+function onMobDisengage(mob)
+    
+    local run = mob:getLocalVar("run")
+    local instance = mob:getInstance()
+    local stage = instance:getStage()
+    local prog = instance:getProgress()
+    
+    if run == 1 then
+        mob:pathThrough(ID.points[stage][prog - 1].point1, 9)
+        mob:setLocalVar("run", 2)
+    elseif run == 2 then
+        mob:pathThrough(ID.points[stage][prog - 1].point2, 9)
+        mob:setLocalVar("run", 3)
+    elseif run == 3 then
+        mob:pathThrough(ID.points[stage][prog - 1].point3, 9)
+        mob:setLocalVar("run", 4)
+    elseif run == 4 then
+        mob:pathThrough(ID.points[stage][prog - 1].point4, 9)
+        mob:setLocalVar("run", 5)
+    elseif run == 5 then
+        mob:pathThrough(ID.points[stage][prog - 1].point5, 9)
+        mob:setLocalVar("run", 6)
+    elseif run == 6 then
+        mob:pathThrough(ID.points[stage][prog - 1].point6, 9)
+        mob:setLocalVar("run", 7)
+    end
+end
+
+function onMobEngaged(mob)
+    mob:setLocalVar("runTime", os.time())
+end
+
+function onMobFight(mob, target)
+    local act = mob:getCurrentAction()
```

Comment:
GetCurrentAction is a deprecated function, just FYI. Hasn't been removed yet but it will be, so we'll need a different way of doing this part

Reply:
Honestly I could probably remove it... it won't interrupt a spell to start running and I think it'd queue up the run for after the spell anyways.

There's a lot I want to rewrite in Arrapago already but I'm tied up with finishing Bhaflau Remnants and getting Silver Seas ready for submission.

Reply:
it can be handled in a separate pr another day, I just wanted to let you know about it